Assets/Shopify/UIToolkit/Components/RemoteImageLoader.cs

```diff
@@ -0,0 +1,77 @@
+namespace Shopify.UIToolkit {
+    using UnityEngine;
+    using UnityEngine.UI;
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// A behaviour that fetches a remote image (JPEG/PNG) from the web.
+    /// </summary>
+    [RequireComponent(typeof(Image))]
+    public class RemoteImageLoader : MonoBehaviour {
+
+        public string ImageURL;
+
+        private Image _image;
+
+        private delegate void RemoteImageCompletionDelegate(Texture2D texture, string error);
+
+        void Start() {
+            _image = gameObject.GetComponent<Image>();
+        } 
+
+        public void LoadImage() {
+            LoadImageURL(ImageURL, (texture, error) => {
+                if (error != null) {
+                    Debug.Log("Failed to download image at " + ImageURL + " Reason: " + error);
```

Comment:
Not in this pr, but we could add a spot to stick a sprite to show when the image is broken or loading in this script too...